Remove the u/b BWC functions.

jamadden opened the pull request:
We don't support 3.2 anymore so they're not needed.

Also remove the BWC import of `ordereddict` since we don't support 2.6 anymore either.

There were essentially no changes in the code proper, almost all the changes were in tests....and doctests. Really glad we're running those on travis now 😄

mgedmin commented on src/zope/schema/_compat.py:
```diff
@@ -2,23 +2,14 @@
-    from collections import OrderedDict
-except ImportError:  # pragma: no cover
-    from ordereddict import OrderedDict
+from collections import OrderedDict
```
We could have modules that use OrderedDict import it directly instead of through `_compat`.

mgedmin commented on src/zope/schema/_compat.py:
```diff
@@ -2,23 +2,14 @@
+from collections import OrderedDict
 
 # pep 8 friendlyness
 OrderedDict
```
That would also make this bit unnecessary.

mgedmin commented on src/zope/schema/tests/test__field.py:
```diff
@@ -48,38 +48,40 @@ def test_validate_wrong_types(self):
-        self.assertRaises(UnicodeEncodeError, byt.fromUnicode, u(chr(129)))
+        u = chr(129)
+        if not isinstance(u, text_type):
+            u = text_type(u, 'unicode_escape') # PY2
```
This seems like a really convoluted way to spell `u'\x81'`.

```
Python 2.7.12 (default, Jul  1 2016, 15:12:24) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> unicode(chr(129), 'unicode_escape')
u'\x81'
```

```
Python 3.5.2 (default, Jul  5 2016, 12:43:10) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> chr(129)
'\x81'
>>> u'\x81'
'\x81'
```

jamadden replied:
Good point, I was just doing a mechanical translation.

mgedmin commented on src/zope/schema/tests/test_equality.py:
```diff
@@ -20,19 +20,14 @@ class FieldEqualityTests(unittest.TestCase):
-        ]
-        for text in equality:
-            self.assertEqual(eval(text), eval(text))
+        for cls in Int,Text:
```
Space after the comma, for PEP-8 friendliness?

jamadden commented:
Hmm, so when you do one of those new-fangled GitHub reviews, when a change is pushed, the comments don't go away ("XXX commented on an outdated diff"), they stay present up front. That would seem to make it harder to see that the changes have been accepted.

freddrake commented:
> Hmm, so when you do one of those new-fangled GitHub reviews, when a change
> is pushed, the comments don't go away ("XXX commented on an outdated diff"),
> they stay present up front. That would seem to make it harder to see that the
> changes have been accepted.

The "outdated" there suggests something relevant to the original comment has changed, but it's impossible to know that the change was driven by the comment (at least without a lot more work & natural language processing), so GitHub is right not to simply hide the comments.

GitLab had been doing the same, but recent versions appear to have declined in the ability to determine the original diff is outdated.

jamadden commented:
The "outdated" applied just to line comments on a line-by-line basis, so *something* on that line clearly changed. Whether it addresses the comment or not is hard to determine, true, but at least there's a way to know that something has changed that is relevant (because it was on the line that had the comment, so its relevant). Now, future viewers of this PR will see all the original comments---which don't show up on the "Files changed" tab because those lines have changed---and unless they manually compare the original commit to the new commit (non-trivial in the Github UI), they won't know that things were addressed on a line-by-line basis (or at least changed).

This is probably all just my version of saying "get off my lawn you kids!" 😄

mgedmin commented:
GitLab has the ability to explicitly mark a comment as resolved, and it hides those by default, like GitHub used to hide comments on outdated diffs.

mgedmin commented on src/zope/schema/tests/test__field.py:
```diff
@@ -72,10 +72,7 @@ def test_validate_required(self):
     def test_fromUnicode_miss(self):
         from zope.schema._compat import text_type
```
This import is unnecessary ;)

mgedmin commented on src/zope/schema/vocabulary.py:
```diff
@@ -21,7 +21,7 @@
 from zope.schema.interfaces import IVocabularyRegistry
 from zope.schema.interfaces import IVocabularyTokenized
-from zope.schema._compat import OrderedDict
+from collections import OrderedDict
```
I'd move this import to the top, but meh.

mgedmin commented on src/zope/schema/tests/test_vocabulary.py:
```diff
@@ -275,7 +275,7 @@ def test_ordering(self):
         #
         #Check that they keys are indeed oredered.
-        from zope.schema._compat import OrderedDict
+        from collections import OrderedDict
```
I'd move this import to the top, but meh.
